src/borg/crypto/key.py

```diff
@@ -442,6 +445,18 @@ def env_passcommand(cls, default=None):
                 raise PasscommandFailure(e)
             return cls(passphrase.rstrip('\n'))
 
+    @classmethod
+    def fd_passphrase(cls):
+        try:
+            fd = int(os.environ.get('BORG_PASSPHRASE_FD', None))
+        except (ValueError, TypeError):
+            return None
+        with os.fdopen(fd) as f:
+            passphrase = f.read()
+        if passphrase is not None:
```

Comment:
passphrase can never be `None` here.

either the `fdopen()` / `read()` blows up with an exception and it'll never get here. (does it make sense to catch it?)

or it works and password will be a `str` (empty or non-empty, we would have to consider both as valid).

Reply:
Leaving out `if'.

If I set BORG_PASSPHRASE_FD and set up the fd, I don't mind borg blowing up if my setup is invalid. At this point it's more UX than anything else.